crmsh/ui_cluster.py

```diff
@@ -91,78 +138,86 @@ def __init__(self):
         self._inventory_target = None
 
     @command.skill_level('administrator')
-    def do_start(self, context):
+    def do_start(self, context, *args):
         '''
-        Starts the cluster services on this node
+        Starts the cluster services on all nodes or specific node
         '''
-        try:
-            if utils.service_is_active("pacemaker.service"):
-                logger.info("Cluster services already started")
-                return
-            bootstrap.start_pacemaker()
-            if utils.is_qdevice_configured():
-                utils.start_service("corosync-qdevice")
-            logger.info("Cluster services started")
-        except IOError as err:
-            context.fatal_error(str(err))
-
-        # TODO: optionally start services on all nodes or specific node
+        node_list = parse_option_for_nodes(context, *args)
+        for node in node_list[:]:
+            if utils.service_is_active("pacemaker.service", remote_addr=node):
+                logger.info("Cluster services already started on {}".format(node))
+                node_list.remove(node)
+        if not node_list:
+            return
+
+        bootstrap.start_pacemaker(node_list)
+        if utils.is_qdevice_configured():
+            utils.start_service("corosync-qdevice", node_list=node_list)
+        for node in node_list:
+            logger.info("Cluster services started on {}".format(node))
 
     @command.skill_level('administrator')
-    def do_stop(self, context):
+    def do_stop(self, context, *args):
         '''
-        Stops the cluster services on this node
+        Stops the cluster services on all nodes or specific node
         '''
-        try:
-            if not utils.service_is_active("corosync.service"):
-                logger.info("Cluster services already stopped")
-                return
-            if utils.service_is_active("corosync-qdevice"):
-                utils.stop_service("corosync-qdevice")
-            utils.stop_service("corosync")
-            logger.info("Cluster services stopped")
-        except IOError as err:
-            context.fatal_error(str(err))
-
-        # TODO: optionally stop services on all nodes or specific node
+        node_list = parse_option_for_nodes(context, *args)
+        for node in node_list[:]:
+            if not utils.service_is_active("corosync.service", remote_addr=node):
+                logger.info("Cluster services already stopped on {}".format(node))
+                node_list.remove(node)
+        if not node_list:
+            return
+
+        # When dlm configured and quorum is lost, before stop cluster service, should set
+        # enable_quorum_fencing=0, enable_quorum_lockspace=0 for dlm config option
```

Comment:
1. Make sense to me to split the code to a separate commit to deal with the dlm node when the cluster is inquorate.
2. also another new commit to add robustness  to do_stop() for https://github.com/ClusterLabs/crmsh/issues/872

Reply:
> > another new commit to add robustness to do_stop() for #872
> @zzhou1 Can't get your point here
I added a simple reproducer in #872

Reply:
```suggestion
```

Reply:
yes, indeed

Reply:
Done